app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -117,55 +145,129 @@ async function walletCreateSessionHandler(
       }
     };
 
+    // if solana is a requested scope but not supported, we add a promptToCreateSolanaAccount flag to request
+    const isSolanaRequested =
+      isNamespaceInScopesObject(
+        requiredScopesWithSupportedMethodsAndNotifications,
+        KnownCaipNamespace.Solana,
+      ) ||
+      isNamespaceInScopesObject(
+        optionalScopesWithSupportedMethodsAndNotifications,
+        KnownCaipNamespace.Solana,
+      );
+
+    let promptToCreateSolanaAccount = false;
+    if (isSolanaRequested) {
+      const supportedSolanaAccounts = await hooks.getNonEvmAccountAddresses(
+        MultichainNetwork.Solana,
+      );
+      promptToCreateSolanaAccount = supportedSolanaAccounts.length === 0;
```

Comment:
Is there a plan for how this could be done chain agnostically?

Reply:
Yeah, probably using the existing [bucketScopes](https://github.com/MetaMask/core/blob/04c89d0fa2d4690e28f7a87cd7553eb5dcd83f5a/packages/chain-agnostic-permission/src/scope/authorization.ts#L70) which can return a list of supportable but not currently supported ScopesObject